pgjdbc/src/test/java/org/postgresql/jdbc/NativeParametersTest.java

```diff
@@ -0,0 +1,421 @@
+/*
+ * Copyright (c) 2022, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.jdbc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.postgresql.PGConnection;
+import org.postgresql.PGPreparedStatement;
+import org.postgresql.PGProperty;
+import org.postgresql.test.TestUtil;
+import org.postgresql.test.jdbc2.BaseTest4;
+import org.postgresql.test.jdbc2.BatchExecuteTest;
+import org.postgresql.util.PSQLException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Properties;
+
+public class NativeParametersTest extends BaseTest4 {
+
+  @Override
+  protected void updateProperties(Properties props) {
+    super.updateProperties(props);
+    PGProperty.PLACEHOLDER_STYLES.set(props, PlaceholderStyles.NATIVE.value());
+  }
+
+  @Test
+  public void dontMix() throws Exception {
+    final PGConnection pgConnection = con.unwrap(PGConnection.class);
+    pgConnection.setPlaceholderStyle(PlaceholderStyles.ANY);
+
+    try {
+      con.prepareStatement("select ?+$1");
+      fail("Should throw a SQLException");
+    } catch (SQLException ex) {
+      // ignore
+      assertEquals(
+          "Multiple bind styles cannot be combined. Saw POSITIONAL first but attempting to also "
+              + "use: NATIVE",
+          ex.getMessage());
+    }
+
+    try {
+      con.prepareStatement("select $1+?");
+      fail("Should throw a SQLException");
+    } catch (SQLException ex) {
+      // ignore
+      assertEquals(
+          "Multiple bind styles cannot be combined. Saw NATIVE first but attempting to also use: "
+              + "POSITIONAL",
+          ex.getMessage());
+    }
+
+    try {
+      con.prepareStatement("select $1+:test");
+      fail("Should throw a SQLException");
+    } catch (SQLException ex) {
+      // ignore
+      assertEquals(
+          "Multiple bind styles cannot be combined. Saw NATIVE first but attempting to also use: "
+              + "NAMED",
+          ex.getMessage());
+    }
+  }
+
+  @Test
+  public void nativeParameterSanity() throws Exception {
+    try {
+      con.prepareStatement("select $1+$4");
+      fail("Should throw a SQLException");
+    } catch (SQLException ex) {
+      // ignore
+      assertEquals(
+          "Native parameter $2 was not found.\n"
+              + "The following parameters where captured: [$1, $4]\n"
+              + "Native parameters must form a contiguous set of integers, starting from 1.",
+          ex.getMessage());
+    }
```

Comment:
There are tests that verify exception messages which is good.
I guess you need to enforce a locale for those tests, otherwise the tests will fail in non-English locale.
At the same time, `try-catch` pattern would better work as `assertThrows` which you use in other tests anyway